batavia/types/Dict.js

```diff
@@ -52,19 +52,30 @@ batavia.types.Dict = function() {
      **************************************************/
 
     Dict.prototype.__lt__ = function(other) {
-        return this.valueOf() < other;
+        throw new batavia.builtins.TypeError("unorderable types: dict() < dict()");
     };
 
     Dict.prototype.__le__ = function(other) {
-        return this.valueOf() <= other;
+        throw new batavia.builtins.TypeError("unorderable types: dict() <= dict()");
     };
 
     Dict.prototype.__eq__ = function(other) {
-        return this.valueOf() == other;
+        if (batavia.isinstance(other, [batavia.types.NoneType]) == true) return false;
```

Comment:
See above about what should happen if the type is not a dict or None.

throwing the `NotImplemented` for the other types is fine here too.